CALCULATOR_UPDATED.c

```diff
@@ -0,0 +1,164 @@
+#include<stdio.h>
+#include<stdlib.h> 
+#include<stdbool.h>
+
+#define OPERATORS "*/+-"  
+
+void readInput(char user_input[]) 
+{
+    printf("Enter a expression to be calculated: ");
+    fgets(user_input,100,stdin);
+    //If i press enter after typing input,it adds \n in string and gives error
+    // here need to replace \n by null  character
+     for (int i = 0; user_input[i] != '\0'; i++)
+    {
+        if (user_input[i] == '\n') 
+        {
+            user_input[i] = '\0';
+            break;
+        }
+    }
+}
+
+
+void removeWhitespaces(char user_input[]) 
+{
+    int iterator_input = 0, iterator_current = 0;
+    while (user_input[iterator_input] != '\0') 
+    {
+        if (user_input[iterator_input] != ' ') 
+        {
+            user_input[iterator_current++] = user_input[iterator_input];
+        }
+        iterator_input++;
+    }
+    user_input[iterator_current] = '\0';
+}
+
+
+bool isDigit(char ch) 

```

Comment:
ch name doesn't make sense and also avoid multiple return statements in the function.